guides-and-tutorials/running-a-signer/how-to-monitor-signer.md

```diff
@@ -0,0 +1,165 @@
+stacks_signer_node_rpc_call_latencies_histogram_bucket{path="/v2/info",le="0.01"} 0
+stacks_signer_node_rpc_call_latencies_histogram_bucket{path="/v2/info",le="0.025"} 0
+stacks_signer_node_rpc_call_latencies_histogram_bucket{path="/v2/info",le="0.05"} 985
+stacks_signer_node_rpc_call_latencies_histogram_bucket{path="/v2/info",le="0.1"} 1194
+# ...
+```
+
+Also, you'll have a `/info` endpoint on the same port:
+
+```bash
+curl 127.0.0.1:30001/info
+```
+
+### Install Alloy
+
+Follow these instructions to install [Grafana
+Alloy](https://grafana.com/docs/alloy/latest/set-up/install/linux/).
+
+On Debian-based distributions:
+
+```bash
+sudo apt install gpg
+sudo mkdir -p /etc/apt/keyrings/
+wget -q -O - https://apt.grafana.com/gpg.key | gpg --dearmor | sudo tee /etc/apt/keyrings/grafana.gpg > /dev/null
+echo "deb [signed-by=/etc/apt/keyrings/grafana.gpg] https://apt.grafana.com stable main" | sudo tee /etc/apt/sources.list.d/grafana.list
+sudo apt-get update
+sudo apt-get install alloy
+```
+
+### Configure Alloy
+
+Edit the file `/etc/alloy/config.alloy` as follows:
+
+```conf
+// For a full configuration reference, see https://grafana.com/docs/alloy
+logging {
+  level = "warn"
+}
+
+prometheus.exporter.unix "default" {
+  include_exporter_metrics = true
+  disable_collectors       = ["mdadm"]
+}
+
+prometheus.scrape "default" {
+  targets = array.concat(
+    prometheus.exporter.unix.default.targets,
+    [
+      {
+        // Self-collect metrics
+        job         = "alloy",
+        __address__ = "127.0.0.1:12345",
+      },
+      {
+        // stacks-signer
+        job         = "stacks-signer",
+        __address__ = "127.0.0.1:30001",
+      },
+      {
+        // stacks-node
+        job         = "stacks-node",
+        __address__ = "127.0.0.1:9153",
+      },
+    ],
+  )
+
+  forward_to = [prometheus.remote_write.default.receiver]
+}
+
+prometheus.remote_write "default" {
+  external_labels = {"instance" = constants.hostname}
+  endpoint {
+    # TODO: Edit the URL below with your Grafana production URL.
+    # should end with /api/prom/push
+    url = "<your GCLOUD_HOSTED_METRICS_URL>"
+
+    # TODO: Edit with your Grafana Cloud ID and Token
+    basic_auth {
+      username = "<your GCLOUD_HOSTED_METRICS_ID>"
+      password = "<your GCLOUD_RW_API_KEY>"
+    }
+  }
+}
+```
+
+```bash
+sudo systemctl daemon-reload
+sudo systemctl enable alloy.service
+sudo systemctl start alloy.service
+```
+
+Metrics from your Signer and node will now start being pushed to Grafana Cloud.
+
+## Visualizing the metrics
+
+You can now start building a dashboard to visualize the metrics.
+
+1. Log-in to Grafana Cloud and create a new Dashboard.
+1. Pick the Prometheus instance you created before as the data source.
+1. Create a new panel and pick `stacks_signer_current_reward_cycle` from the
+  metrics.
+
+You should now be able to see Stacks' current reward cycle, as measured by the
+Signer, into the dashboard.
```

Comment:
I would add a final link to either the promql docs, or grafana's documentation: https://grafana.com/docs/grafana-cloud/visualizations/panels-visualizations/query-transform-data/

https://prometheus.io/docs/prometheus/latest/querying/basics/

since the metrics are important, but knowing how to query them for averages (etc), and combining metrics is quite important here.

Reply:
Thanks, addressed in ac4d1cfa9